### PR TITLE
Rename the wasmtime_api library to match the containing wasmtime crate

### PR DESCRIPTION
**Pull request by joshtriplett**
Commit d9ca508f8055ae273c9adcc114181e5782900a1d renamed the `wasmtime-api` crate to `wasmtime`, but left the name of the library it contains as `wasmtime_api`.

It's fairly unusual for a crate to contain a library with a different name, and it results in rather confusing error messages for a user; if you list `wasmtime = "0.7"` in `Cargo.toml`, you can't `use wasmtime::*`, you have to `use wasmtime_api::*;`.

(Also, docs.rs says that [wasmtime-0.7.0 is not a library](https://docs.rs/crate/wasmtime/0.7.0).)

Is there a strong reason we *shouldn't* rename the `wasmtime_api` library to `wasmtime`, to match the crate name? I'd be happy to submit a PR to that effect.

### CODE REVIEW

**Comment by joshtriplett** [comment]
Rebased. Couldn't test the new version, as witx currently has a build failure.

File: crates/wasi/src/instantiate.rs [hunk 1]
```diff
@@ -9,20 +9,19 @@ use std::fs::File;
...
+    store: &wasmtime::HostRef<wasmtime::Store>,
     preopened_dirs: &[(String, File)],
     argv: &[String],
     environ: &[(String, String)],
-) -> Result<api::Instance, InstantiationError> {
+) -> Result<wasmtime::Instance, InstantiationError> {
     let global_exports = store.borrow().global_exports().clone();
     let wasi = instantiate_wasi(global_exports, preopened_dirs, argv, environ)?;
-    let instance = api::Instance::from_handle(&store, wasi);
+    let instance = wasmtime::Instance::from_handle(&store, wasi);
```
Comment (kubkon) [inline crates/wasi/src/instantiate.rs]: If you rebase to the latest `master` and apply the same changes to `crates/wasi/src/old/snapshot_0/instantiate.rs`, the CI should work just fine. I don't think there is anything wrong with `witx` as such.

Comment (joshtriplett) [reply crates/wasi/src/instantiate.rs]: The build never got this far, which is why I didn't see this. Thank you for catching this, I'll update.

I don't think this relates to the issue with `witx`, which seems unrelated:

```
error[E0599]: no method named `comment` found for type `wast::parser::Cursor<'_>` in the current scope
   --> crates/wasi-common/WASI/tools/witx/src/parser.rs:111:49
    |
111 |                 let (comment, c) = match cursor.comment() {
    |                                                 ^^^^^^^ method not found in `wast::parser::Cursor<'_>`
```

**Comment by joshtriplett** [comment]
I managed to track down the build failure with witx:

witx depends on `wast = "3.0.1"`, but it uses the `comment()` method of the `Cursor` type, which doesn't exist in version 3.0.1. It exists in the current 3.0.4. So, witx needs an updated dependency. (I can work around this problem by running `cargo update`, but witx needs a dependency fix.)

File: crates/wasi/src/instantiate.rs (hunk 1 above)
Comment (kubkon) [reply crates/wasi/src/instantiate.rs]: Ah, I see. Somehow I missed that error, thanks for pointing it out! :-)

**Comment by sunfishcode** [comment]
Ah, thanks for tracking that down, and fixing it!

**Pull request merged**